clippy_lints/src/methods/bytes_nth.rs

```diff
@@ -16,18 +18,21 @@ pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &Expr<'_>, recv: &'tcx E
     } else {
         return;
     };
+
     let mut applicability = Applicability::MachineApplicable;
-    span_lint_and_sugg(
-        cx,
-        BYTES_NTH,
-        expr.span,
-        &format!("called `.bytes().nth()` on a `{caller_type}`"),
-        "try",
-        format!(
-            "{}.as_bytes().get({})",
+    if let Some(parent) = clippy_utils::get_parent_expr(cx, expr)
+      && let Some((name, _, _, _, _)) = method_call(parent)
+      && name == "unwrap" {
+        span_lint_and_sugg(cx, BYTES_NTH, parent.span, &format!("called `.bytes().nth().unwrap()` on a `{caller_type}`"), "try", format!(
+            "{}.as_bytes()[{}]",
+            snippet_with_applicability(cx, recv.span, "..", &mut applicability),
+            snippet_with_applicability(cx, n_arg.span, "..", &mut applicability)
```

Comment:
You could probably fetch the snippets directly after defining `applicability`, so you don't need to do it twice.